### ISSUE TITLE
Z and M dimension support

### ISSUE THREAD

**Issue by evetion** [yeesian/ArchGDAL.jl#291]
While going through #290, I noticed some issues with geometry types, especially the non-default ones with Z and/or M dimensions.

- Missing `getm`. (added in #290)
- Creation of geometries don't support m (four coordinates)
- Creation of geometries don't set correct type, createpoint(0,0) and createpoint(0,0,0) both result in `ArchGDAL.IGeometry{ArchGDAL.wkbPoint}`, while the latter should be `wkbPoint25D` or `wkbPointM`.

**Comment by yeesian** [comment]
how do we know whether `createpoint(0,0,0)` should be `wkbPoint25D` or `wkbPointM`?

**Comment by yeesian** [comment]
My implicit understanding was that `createpoint(0,0,0)` would have PointZ semantics, and PointM and Point25D are not supported in this package yet.

**Comment by rafaqz** [comment]
@yeesian I agree, regular arguments should be spatial dimensions. M could use a keyword argument.

**Comment by yeesian** [comment]
Does that mean
* `createpoint(0,0)` -> `ArchGDAL.IGeometry{ArchGDAL.wkbPoint}`
* `createpoint(0,0,0)` -> `ArchGDAL.IGeometry{ArchGDAL.wkbPoint25D}` (per https://gis.stackexchange.com/a/7869)
* `createpoint(0,0,m=0)` -> `ArchGDAL.IGeometry{ArchGDAL. wkbPointM}`
* `createpoint(0,0,0,m=0)` -> `ArchGDAL.IGeometry{ArchGDAL.wkbPointZM}`

?

**Comment by rafaqz** [comment]
Yeah, that would work.

**Comment by evetion** [comment]
Discussed `is3d` and `ismeasured` in #303, which are now implemented in #290. Methods to actually create/set these coordinates (especially for `m`) are still missing.
